pkg/sdkserver/sdkserver.go

```diff
@@ -491,14 +492,38 @@ func (s *SDKServer) GetGameServer(context.Context, *sdk.Empty) (*sdk.GameServer,
 		return nil, err
 	}
 
-	return convert(gs), nil
+	s.cachedGameServer = convert(gs)
+	return s.cachedGameServer, nil
+}
+
+func (s *SDKServer) getCachedGameServer() (*sdk.GameServer, error) {
+	if s.cachedGameServer != nil {
+		return s.cachedGameServer, nil
+	}
+
+	return s.GetGameServer(context.Background(), &sdk.Empty{})
 }
 
 // WatchGameServer sends events through the stream when changes occur to the
 // backing GameServer configuration / status
 func (s *SDKServer) WatchGameServer(_ *sdk.Empty, stream sdk.SDK_WatchGameServerServer) error {
 	s.logger.Debug("Received WatchGameServer request, adding stream to connectedStreams")
 	s.streamMutex.Lock()
+
+	if runtime.FeatureEnabled(runtime.FeatureSDKWatchSendOnExecute) {
+		gs, err := s.getCachedGameServer()
```

Comment:
Not sure why we have this `getCacheGameServer()` - the Informer always has the up to date GameServer cached in memory, so we don't need to double-cache it I don't think, and it's likely a recipe for race conditions.